botocore/stub.py

```diff
@@ -265,7 +314,21 @@ def _get_response_handler(self, model, params, **kwargs):
     def _assert_expected_params(self, model, params, **kwargs):
         self._assert_expected_call_order(model, params)
         expected_params = self._queue[0]['expected_params']
-        if expected_params is not None and params != expected_params:
+        if expected_params is None:
+            return
+
+        # Validate the parameters are equal
+        for param, value in expected_params.items():
+            if value is ANY:
+                continue
+            elif expected_params[param] != params[param]:
```

Comment:
I feel like there may be key errors here if the expected param is missing from the actual params.